paasta_tools/marathon_tools.py

```diff
@@ -377,10 +377,19 @@ def format_marathon_app_dict(self):
             complete_config['container']['docker']['portMappings'] = [
                 {
                     'containerPort': CONTAINER_PORT,
-                    'hostPort': 0,
+                    'hostPort': self.get_host_port(),
                     'protocol': 'tcp',
                 },
             ]
+        else:
+            complete_config['port_definitions'] = [
+                {
+                    'port': self.get_host_port(),
+                    'protocol': 'tcp',
+                },
+            ]
+            # Without this, we may end up with multiple containers requiring the same port on the same box.
+            complete_config['require_ports'] = True
```

Comment:
Actually, that's an interesting point. We might be able to do without it, so long as we add hostname:UNIQUE constraints.

requirePorts does restrict us to only using ports that are in the mesos offers (e.g. 31000-31999).

Should I drop it?